driver/src/main/com/mongodb/FindIterableImpl.java

```diff
@@ -34,20 +35,23 @@
 import static com.mongodb.assertions.Assertions.notNull;
 import static java.util.concurrent.TimeUnit.MILLISECONDS;
 
-final class FindIterableImpl<T> implements FindIterable<T> {
+final class FindIterableImpl<TResult, TDocument> implements FindIterable<TResult> {
     private final MongoNamespace namespace;
-    private final Class<T> clazz;
+    private final Class<TResult> clazz;
     private final ReadPreference readPreference;
     private final CodecRegistry codecRegistry;
     private final OperationExecutor executor;
     private final FindOptions findOptions;
-    private Object filter;
+    private final Class<TDocument> collectionClass;
```

Comment:
I'm not sure how `collectionClass` is used which lead me to double check the `Renderable` use of it - is it needed?